pandas/io/formats/style.py

```diff
@@ -1222,6 +1222,76 @@ class MyStyler(cls):
 
         return MyStyler
 
+    def pipe(self, func, *args, **kwargs):
+        """
+        Apply func(self, \*args, \*\*kwargs), and return the result.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        func : function
+            function to apply to the Styler.
+            ``args``, and ``kwargs`` are passed into ``func``.
+            Alternatively a ``(callable, data_keyword)`` tuple where
+            ``data_keyword`` is a string indicating the keyword of
+            ``callable`` that expects the Styler.
+        args : iterable, optional
+            positional arguments passed into ``func``.
+        kwargs : mapping, optional
+            a dictionary of keyword arguments passed into ``func``.
+
+        Returns
+        -------
+        object : the value returned by ``func``.
+
+        See Also
+        --------
+        Styler.apply
+        Styler.applymap
+        pandas.DataFrame.pipe
+
+        Notes
+        -----
+        Like :meth:`pandas.DataFrame.pipe`, this method can simplify the
+        application of several user-defined functions to a styler.  Instead
+        of writing:
+
+        >>> f(g(df.style.set_precision(3), arg1=a), arg2=b, arg3=c)
+
+        users can write:
+
+        >>> df.style.set_precision(3)
+        ...     .pipe(g, arg1=a)
+        ...     .pipe(f, arg2=b, arg3=c)
+
+        In particular, this allows users to define functions that take a
+        styler object, along with other parameters, and return the styler after
+        making styling changes (such as calling :meth:`Styler.apply` or
+        :meth:`Styler.set_properties`).  Using ``.pipe``, these user-defined
+        style "transformations" can be interleaved with calls to the built-in
+        Styler interface.
+
+        Examples
+        --------
+        >>> def highlight_with_containing_row(styler, row_label, column):
+        ...     ''' Highlight the indicated element, and its containing row. '''
+        ...     cell, row = pd.IndexSlice[row_label, column], pd.IndexSlice[row_label, :]
+        ...     return (styler.set_properties(cell, **{'background-color': '#ffffcc'})
+        ...                   .set_properties(row, **{'background-color': '#ffccc0'}))
```

Comment:
It'd be great if you can find a simpler example to illustrate `.pipe()`. Something like adding quotes around the values, or something as simple as that, so the function does not distract on what `.pipe()` does.

Reply:
That is a good suggestion.  I chose this to illustrate functionality that seems to belong in a function, and to use a function that could sensibly be applied twice.

BUT that's not the clearest example for the docs.  I'll use a much simpler function as you suggest.